src/snackbar.vue

```diff
@@ -6,11 +6,23 @@
 
 <script>
 export default {
-  props: ['displayOn'],
-  ready () {
+  props: {
+    displayOn: {
+      type: String,
+      required: true
+    },
+    eventSource: {
+      type: Object,
+      required: false,
+      default: function () {
+        return this.$root
+      }
+    }
+  },
+  mounted () {
     componentHandler.upgradeElement(this.$el, 'MaterialSnackbar')
-    this.$on(this.displayOn, (snackarConfig) => {
-      this.$el.MaterialSnackbar.showSnackbar(snackarConfig)
+    this.eventSource.$on(this.displayOn, (snackbarConfig) => {
```

Comment:
Because Vue events no longer propagate down or bubble up, The snackbar now needs to listen for events on a specific event emitter. In order to have a simple default, the property default for the source is the `$root`, which lets anyone invoke the snackbar just using `$root.$emit('mailSent', { message: 'Message Sent' })`

However, if you are using a global event bus, you might want to use that as the event emitter to listen on, in which case you could bind the `event-source` property appropriately.

I showed examples of this in the snackbar test.